File: app/src/main/java/com/arslan/guard/AppInfo.java
package com.arslan.guard;

import android.graphics.drawable.Drawable;

/**
 * Cihazda yüklü tek bir uygulamayı temsil eden basit model sınıfı.
 */
public class AppInfo {

    private final String appName;
    private final String packageName;
    private final Drawable icon;
    private boolean locked;

    public AppInfo(String appName, String packageName, Drawable icon, boolean locked) {
        this.appName = appName;
        this.packageName = packageName;
        this.icon = icon;
        this.locked = locked;
    }

    public String getAppName() {
        return appName;
    }

    public String getPackageName() {
        return packageName;
    }

    public Drawable getIcon() {
        return icon;
    }

    public boolean isLocked() {
        return locked;
    }

    public void setLocked(boolean locked) {
        this.locked = locked;
    }
}

File: app/src/main/java/com/example/applock/AppInfo.kt
package com.example.applock

import android.graphics.drawable.Drawable

/**
 * Uygulama listesinde gösterilen tek bir yüklü uygulamayı temsil eder.
 */
data class AppInfo(
    val packageName: String,
    val appName: String,
    val icon: Drawable,
    var isLocked: Boolean
)
